### ISSUE TITLE
Does this work in Windows with "git for Windows"

### ISSUE THREAD

**Issue by back5576** [stevelacy/gulp-git#78]
Took inspiration from:

http://ponyfoo.com/articles/my-first-gulp-adventure

and tried this (just the checkin part for starters):
```js
// tag and checkin to git
gulp.task('tag', function () {
  var pkg = require('./package.json');
  var v = 'v' + pkg.version;
  var message = 'Release ' + v;
  return gulp.src('./')
    .pipe(git.commit(message));
});
```
and get error:

C:\eac>gulp tag
[21:26:25] Using gulpfile C:\eac\gulpfile.js
[21:26:25] Starting 'tag'...
[21:26:25] 'tag' errored after 46 ms
[21:26:25] Error: Command failed:
    at ChildProcess.exithandler (child_process.js:648:15)
    at ChildProcess.emit (events.js:98:17)
    at maybeClose (child_process.js:756:16)
    at Process.ChildProcess._handle.onexit (child_process.js:823:5)

which is Greek to me since I'm not a "gulp developer".
Does gulp-git work on Windows?

**Comment by stevelacy** [comment]
Do you have `git` installed? Can you run the command `git` and return the help usage of git?

That error is not gulp, it is a node.js stack trace.

**Comment by back5576** [comment]
I have this git installed: https://msysgit.github.io/
Here's some more info
C:\Users\chris>which git.exe
C:\Program Files (x86)\Git\bin\git.exe

C:\Users\chris>where git
C:\Program Files (x86)\Git\bin\git.exe
C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Web Tools\External\git\git.exe

C:\Users\chris>git help
usage: git [--version] [--help] [-C <path>] [-c name=value]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p|--paginate|--no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           <command> [<args>]

The most commonly used git commands are:
   add        Add file contents to the index
   bisect     Find by binary search the change that introduced a bug
   branch     List, create, or delete branches
   checkout   Checkout a branch or paths to the working tree
   clone      Clone a repository into a new directory
   commit     Record changes to the repository
   diff       Show changes between commits, commit and working tree, etc
   fetch      Download objects and refs from another repository
   grep       Print lines matching a pattern
   init       Create an empty Git repository or reinitialize an existing one
   log        Show commit logs
   merge      Join two or more development histories together
   mv         Move or rename a file, a directory, or a symlink
   pull       Fetch from and integrate with another repository or a local branch
   push       Update remote refs along with associated objects
   rebase     Forward-port local commits to the updated upstream head
   reset      Reset current HEAD to the specified state
   rm         Remove files from the working tree and from the index
   show       Show various types of objects
   status     Show the working tree status
   tag        Create, list, delete or verify a tag object signed with GPG

'git help -a' and 'git help -g' lists available subcommands and some
concept guides. See 'git help <command>' or 'git help <concept>'
to read about a specific subcommand or concept.

Please let me know if this gives you any ideas, or if there is more info about my environment that may shed light.

 - Chris

**Comment by stevelacy** [comment]
Can you create a simple test repo to reproduce the bug?

I'll need to find a way to test this on windows...

**Comment by back5576** [comment]
I'm working on a different project this week. I'll try to create a simple test next week.
 - Thanks, Chris

**Comment by legacybass** [comment]
Not sure if this has been resolved or not, but thought I'd add my two cents. I have git installed from the same msysgit website, and have had no problems. I've written our entire build server using gulp, and a large portion of this is using git.

I tested the above code to see if it works, and I'm able to run it and see the commit with no problems. Not sure if it's a bug in the rest of the code/setup, but the snippet shared seems to work without issue on Windows (at least on Windows 10 and 8).
